src/Mvc/Mvc.Core/src/Infrastructure/DefaultProblemDetailsFactory.cs

```diff
@@ -9,20 +9,37 @@
 using Microsoft.Extensions.Options;
 
 namespace Microsoft.AspNetCore.Mvc.Infrastructure;
-
-internal sealed class DefaultProblemDetailsFactory : ProblemDetailsFactory
+/// <summary>
+/// The `DefaultProblemDetailsFactory` is a concrete implementation of the `ProblemDetailsFactory` abstract class.
+/// It provides methods to create instances of `ProblemDetails` and `ValidationProblemDetails` with default settings.
+/// This class uses the provided `ApiBehaviorOptions` for client error mapping and an optional custom configuration action to further customize the problem details.
+/// </summary>
+public sealed class DefaultProblemDetailsFactory : ProblemDetailsFactory
```

Comment:
@GiovanniBraconi please address the analyzer-failure.
Have a look at [How to use Microsoft.CodeAnalysis.PublicApiAnalyzers](https://github.com/dotnet/roslyn-analyzers/blob/main/src/PublicApiAnalyzers/PublicApiAnalyzers.Help.md) and see at the files in https://github.com/dotnet/aspnetcore/tree/main/src/Mvc/Mvc.Core/src folder.